java/src/jmri/util/swing/JmriMouseEvent.java

```diff
@@ -0,0 +1,483 @@
+package jmri.util.swing;
+
+import java.awt.Component;
+import java.awt.Point;
+import java.awt.Toolkit;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseEvent;
+
+import javax.swing.SwingUtilities;
+
+import jmri.util.SystemType;
+
+/**
+ * Adaptor for MouseEvent.
+ * This class is used to fix some issues with MouseEvent on Windows.
+ *
+ * @author Daniel Bergqvist (C) 2022
+ */
+public class JmriMouseEvent {
```

Comment:
Could this inherit from MouseEvent?  Or are there technical issues (final, etc) that make that impossible?  (Just curious)

Reply:
Yes, I think it could. At least I don't know of any reason for why it couldn't be done.

There was two reasons to why I designed it this way:

1. I wanted compile time errors for the cases I didn't yet had converted. For me, the easiest way to do these kind of changes is to change at one place and then handle all the compile time errors that follows.

2. It forces other developers to use JmriMouseEvent instead of using MouseEvent. It might seem harsh to think that way, but the problem is that the differences in for example `isPopupTrigger()` might cause bugs that are very difficult to detect. To try to prevent that to happen again, I think it's better to force the use of JmriMouseEvent. Since JmriMouseEvent doesn't inherit MouseEvent, both the caller and the called method must use JmriMouseEvent. That helps to keep JMRI using this pattern which in turn keeps JMRI away from these bugs.

So this is by design only. If you prefer `JmriMouseEvent` to inherit `MouseEvent`, I'm happy to do that. But in the long run I think it's better to keep it as is.

Reply:
Thank you!.  That makes a lot of sense.